### PR TITLE
Defer to `edit_posts` of the custom post type

### PR DESCRIPTION
**Pull request by danielbachhuber**
These private query variables should be accessible to CPTs when a user
can't create normal posts.

### CODE REVIEW

**Comment by danielbachhuber** [comment]
Need to add some tests around private query var handling.

**Comment by rmccue** [comment]
:+1: on this, good spot.

**Comment by danielbachhuber** [comment]
@WP-API/amigos #reviewmerge

File: tests/test-rest-pages-controller.php [hunk 1]
```diff
@@ -47,6 +47,25 @@ public function test_get_items() {
 
 	}
 
+	public function test_get_items_private_filter_query_var() {
+		// Private query vars inaccessible to unauthorized users
+		wp_set_current_user( 0 );
```
Comment (danielbachhuber) [inline tests/test-rest-pages-controller.php]: I wanted to test contributors here too, as they don't have the ability to edit pages, but I ran into #1892

**Comment by joehoyle** [comment]
:+1:

**Pull request merged**